### PR TITLE
Friendlier and cleaner repack.sh

### PR DESCRIPTION
**Pull request by hackerb9**
Make the repack script friendlier to users by simplifying the usage syntax.
*  Emit more helpful error messages.
*    Embed template file in script so it can run without any external dependencies.
 *  On success, tell the user what the logical next step would be.
 * Clean up, per suggestions in #32.
 * Additional shell script simplification and improvement.

Closes https://github.com/linux-surface/surface-uefi-firmware/issues/31 .

### CODE REVIEW

File: repack.sh [hunk 1]
```diff
@@ -163,40 +224,68 @@ repackmsi()
...
 	# Extract the CAB
-	TEMP="$(mktemp -p . -d)"
+	local TEMP="$(mktemp -p . -d)"
 	gcab -C "${TEMP}" -x "${CAB}" > /dev/null
 
+	# Repack all UEF capsule updates found in the CAB
+	local inffiles=($(grep -lR 'Firmware_Install,UEFI' "${TEMP}"))
 	# Convert all .inf files to unix format
-	find "${TEMP}" -iname '*.inf' -exec sh -c 'dos2unix "$0" > /dev/null 2>&1' {} \;
+	sed -i 's/\r//' "${inffiles[@]}"
```
Comment (StollD) [inline repack.sh:239]: It still tests for dos2unix above and fails if its not present FYI.

I also think there actually were encoding issues in some of the files that dos2unix managed to fix. But maybe that was just my editor going crazy on carriage returns at the time?

How about testing if dos2unix is available, and falling back to sed if its not?

File: repack.sh [hunk 2]
```diff
@@ -1,81 +1,139 @@
...
+	   fi
+    esac
+
+    if [[ ${#CAB_ARRAY[@]} -gt 0 ]]; then
+	echo "Success!"
+	echo "If you wish, you may now install the firmware like so:"
+	echo
+	local f
+	for f in "${CAB_ARRAY[@]}"; do
+	    echo -n "  sudo fwupdmgr install --allow-older --no-reboot-check --force "
```
Comment (StollD) [inline repack.sh:110]: The README has --allow-reinstall but this doesn't, is that intentional?

File: repack.sh (hunk 1 above)
Comment (hackerb9) [reply repack.sh:239]: While dos2unix can modify the encoding, the manpage for the Linux version says that the default is "ASCII conversion", which converts only line breaks. (Technically, dos2unix for SunOS/Solaris behaves differently as it converts the encoding to ISO 8859-1 --- which is unlikely to help anything these days). 

I suggest removing dos2unix completely. Let me know if that's okay and I'll make the change.

File: repack.sh (hunk 2 above)
Comment (hackerb9) [reply repack.sh:110]: Not intentional. Will fix.

File: repack.sh (hunk 1 above)
Comment (hackerb9) [reply repack.sh:239]: Actually, scratch that. It also converts UTF-16 to UTF-8, so it is definitely necessary. I'll put it back in.

Comment (hackerb9) [reply repack.sh]: Okay, I've put dos2unix back into the script (albeit in a tad nicer syntax).

**Comment by hackerb9** [comment]
Okay, I think I've got all the requested changes in place.

**Review by StollD — APPROVED** [review]

**Comment by StollD** [comment]
Thank you!

**Pull request merged**